packages/contracts/contracts/governance/GovernanceToken.sol

```diff
@@ -1,47 +1,113 @@
+        require(
+            msg.sender == BRIDGE || msg.sender == MINT_MANAGER,
+            "GovernanceToken: only minter can mint"
+        );
+
         _mint(_account, _amount);
+        emit Mint(_account, _amount);
+    }
+
+    /**
+     * @notice Returns the total minted amount.
+     *
+     * @return The total minted amount.
+     */
+    function totalMinted() public view returns (uint256) {
+        return _totalMinted;
+    }
+
+    /**
+     * @notice Returns the maximum number of tokens that can be minted.
+     *
+     * @return The maximum number of tokens.
+     */
+    function cap() public pure returns (uint256) {
+        return MAX_TOTAL_SUPPLY;
     }
 
-    /// @notice Callback called after a token transfer.
-    /// @param from   The account sending tokens.
-    /// @param to     The account receiving tokens.
-    /// @param amount The amount of tokens being transfered.
-    function _afterTokenTransfer(address from, address to, uint256 amount) internal override(ERC20, ERC20Votes) {
-        super._afterTokenTransfer(from, to, amount);
+    /**
+     * @notice Callback called after a token transfer.
+     *
+     * @param _from   The account sending tokens.
+     * @param _to     The account receiving tokens.
+     * @param _amount The amount of tokens being transferred.
+     */
+    function _afterTokenTransfer(
+        address _from,
+        address _to,
+        uint256 _amount
+    ) internal override(ERC20, ERC20Votes) {
+        super._afterTokenTransfer(_from, _to, _amount);
     }
 
-    /// @notice Internal mint function.
-    /// @param to     The account receiving minted tokens.
-    /// @param amount The amount of tokens to mint.
-    function _mint(address to, uint256 amount) internal override(ERC20, ERC20Votes) {
-        super._mint(to, amount);
+    /**
+     * @notice Internal mint function.
+     *
+     * @param _account The account receiving minted tokens.
+     * @param _amount  The amount of tokens to mint.
+     */
+    function _mint(address _account, uint256 _amount) internal override(ERC20, ERC20Votes) {
+        uint256 minted = _totalMinted + _amount;
+        require(minted <= MAX_TOTAL_SUPPLY, "GovernanceToken: cap exceeded");
+        super._mint(_account, _amount);
+        _totalMinted = minted;
+        emit Mint(_account, _amount);
     }
 
-    /// @notice Internal burn function.
-    /// @param account The account that tokens will be burned from.
-    /// @param amount  The amount of tokens that will be burned.
-    function _burn(address account, uint256 amount) internal override(ERC20, ERC20Votes) {
-        super._burn(account, amount);
+    /**
+     * @notice Internal burn function.
+     *
+     * @param _account The account that tokens will be burned from.
+     * @param _amount  The amount of tokens that will be burned.
+     */
+    function _burn(address _account, uint256 _amount) internal override(ERC20, ERC20Votes) {
```

Comment:
Wouldn't we need sender check?

Reply:
This function calls `ERC20._burn()`, so there's no need for a sender check.